Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.tsx

```diff
@@ -16,61 +18,198 @@ import styles from './ContestCategories.module.scss';
 interface IContestCategoriesProps extends IHaveOptionalClassName {
     onCategoryClick: (filter: IFilter) => void;
     defaultSelected?: string;
+    setStrategyFilters: Dispatch<SetStateAction<IFilter[]>>;
 }
 
+interface IFilterProps {
+    categoryId: string;
+    strategies: IFilter[];
+}
+
+const defaultState = {
+    state: {
+        openedCategoryFilters: [] as IFilterProps[],
+        openedCategoryFilter: {} as IFilterProps,
+    },
+};
 const ContestCategories = ({
     className = '',
     onCategoryClick,
     defaultSelected,
+    setStrategyFilters,
 }: IContestCategoriesProps) => {
     const { state: { categories } } = useContestCategories();
     const { state: { possibleFilters } } = useContests();
-    const { actions: { updateBreadcrumb } } = useCategoriesBreadcrumbs();
+    const { actions: { updateBreadcrumb, clearBreadcrumb } } = useCategoriesBreadcrumbs();
+    const [ openedStrategyFilters, setOpenedStrategyFilters ] = useState(defaultState.state.openedCategoryFilters);
+    const [ openedStrategyFilter, setOpenedStrategyFilter ] = useState(defaultState.state.openedCategoryFilter);
+    const [ currentCategoryId, selectCurrentCategoryId ] = useState<string>('');
+    const [ prevCategoryId, setPrevCategoryId ] = useState<string>('');
+
+    const categoriesFlat = useMemo(
+        () => flattenWith(categories, (c) => c.children || null),
+        [ categories ],
+    );
+
+    const getCategoryId = useCallback(
+        (searchedId?: string, searchedValue?: string) => {
+            const filterToFind = possibleFilters
+                .find(({ id, value }) => id.toString() === searchedId || value === searchedValue?.toString()) as IFilter;
+
+            if (isNil(filterToFind)) {
+                return '';
```

Comment:
Isnt the id a number?

Reply:
Yes, but we need to convert it to string in order to work with the searchedId.